### PR TITLE
Optional requirements in `setup.py`

### PR DESCRIPTION
**Pull request by bersbersbers**
Thank your for contributing to Nuitka!

# What does this PR do?

Enable `pip install -e .[all]`, for instance, to install optional dependencies. Names of dependencies TBC, this is more a technical demo.

# Why was it initiated? Any relevant Issues?

#2053

Requirements were pulled from `requirements.txt` and diffed against existing ones in `setup.py` - missing ones added as optional.

# PR Checklist

- [x] Correct base branch selected? Should be `develop` branch.
- [x] Enabled commit hook or executed `./bin/autoformat-nuitka-source`.
- [ ] All tests still pass. Check the Developer Manual about `Running the Tests`.
      There are GitHub Actions tests that cover the most important
      things however, and you are welcome to rely on those, but they might not
      cover enough.
- [ ] Ideally new features or fixed regressions ought to be covered via new tests.
- [ ] Ideally new or changed features have documentation updates.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -430,4 +430,14 @@ def has_ext_modules():
...
     # wheel version and platform specific.
     distclass=BinaryDistribution,
+    extras_require={
+        "all": [
+            "ordered-set; python_version >= '3.10'",
+            "subprocess32; python_version == '2.7'",
+            "zstandard; python_version >= '3.5'",
+        ],
+        "msvc": [
+            "orderedset; python_version >= '3.6' or os.name != 'nt'",
```
Comment (kayhayen) [inline setup.py:440]: For orderedset, it's way more of a mess. Should be more like this:

```
orderedset == 2.0.3 ; os.name != "nt" and sys.platform != "darwin" and python_version < "3.7"
orderedset == 2.0.3 ; os.name == "nt" and python_version >= '3.4' and python_version < "3.7"
orderedset == 2.0.3 ; sys.platform == "darwin" and python_version < "3.7"
ordered-set == 4.1.0; python_version >= "3.7"
```

and it's not for MSVC, but I think I get the idea.

**Comment by kayhayen** [comment]
So, we could offer these names for installation, and `all` would install the recommended dependencies, maybe later e.g. also what it takes to check the schema of user yaml files. To avoid conflicts, I would want to have the versions be more of the `>=` kind rather than `==` or else it may create conflicts.

I will trry it with 1.5 release, but keep it unofficial until I tried it out. I don't think that keeping `onefile` separate is useful, so lets start with `all`.

I think I can take it from here, not sure how useful it is to do it in a PR now.

File: setup.py (hunk 1 above)
Comment (bersbersbers) [reply setup.py:440]: Sure! (I just copy-and-pasted blindy from `requirements.txt`.)

**Comment by bersbersbers** [comment]
> To avoid conflicts, I would want to have the versions be more of the `>=` kind

Yes!

> I will trry it with 1.5 release

Cool!

> I think I can take it from here, not sure how useful it is to do it in a PR now.

Agreed - continue as it suits you, and feel free to close this PR.

**Comment by kayhayen** [comment]
Ultimately I think I realized, I just should have `install_requires` properly and if people not want them, they should tell pip about that, a `all` only expansion would suffer from most people forever being unaware of it, and having dependencies is not like unhear of.

**Comment by bersbersbers** [comment]
Fully agree. Just note it can hard to tell pip to ignore a dependency (https://github.com/pypa/pip/issues/8076), so people would have to uninstall after the fact (assuming they were able to install in the first place).

**Pull request closed without merging**